### DISCUSSION TITLE
GUI crashes after calibration with certain motors

### DISCUSSION THREAD

**Discussion by AuroraD-Hub** [tinymovr/Tinymovr#330] [Q&A - Tinymovr] [answered]
# Final goal
I want to control a 3-phase brushless motor (**[AK 80-6 KV100](https://store.tmotor.com/product/ak80-6-dynamical-modular.html)**) in velocity mode with a position feedback with **R5.2** or **M5.2**.

# Context
I am testing the boards with the Tinymovr GUI provided by first performing calibration procedure and then sending velocity or position setpoints.  
I'm using the following different motors:  
- [AK 80-6 KV100](https://store.tmotor.com/product/ak80-6-dynamical-modular.html) with 42 poles and reduction ratio 6:1 **PROBLEMATIC**;
- [MN6007 KV320](https://store.tmotor.com/product/mn6007-kv160-motor-antigravity-type.html) with 22 poles (no reduction ratio)**OK**;
- [U8 II KV85](https://store.tmotor.com/product/u8-v2-u-efficiency-kv85.html) with 42 poles (no reduction ratio)**PROBLEMATIC**.

# Problem
When I perform the calibration of the **AK 80-6 KV100** or **U8 II KV85** motor in the GUI the flag *calibrated* in the *motor* section is *True*, but the same flag in the *encoder* and *tmx* section are *False*. Moreover, the GUI window freezes and it seems the Tinymvor is not visible in the GUI anymore when the calibration is done. Then I have to restart the Tinymovr to make it visible to the GUI. Still, in the  *motor* section the parameters are properly calibrated after restarting the motor, but not in the *encoder* and *tmx* sections.

To understand where the problem is, I firstly used the **MN6007 KV320** motor with both the R5.2 and M5.2 Tinymovr boards and I managed to successfully perform the calibration procedure and control the motor discarding possible faults of the boards. With the MN6007 everything works as expected. Then, I used the **U8 II KV85** with R5.2 and got the same problems I had with the **AK 80-6 KV100**.  

# Question
May it be possible that these Tinymovr boards are able to control only motors under a certain number of poles? Since both the AK 80-6 KV100 and the U8 II KV85 have a relatively large number of poles (21 pole pairs).

**Comment by yconst** [comment, marked as answer]
This is an issue on our part, as the pole pairs are limited to 20 in 1.6.x this has been raised to 24 in 1.7, and were also adding a specific error to be displayed if the pole pairs count is greater than that.

Not sure about the gui freeze though, does the terminal show anything?

**Reply by AuroraD-Hub** [reply]
Thank you very much!
Can I ask whenever the new firmware version will be released?

Regarding the GUI freezing, it restarted at some point and just failed when using not supported motors. So this is not a problem I think.

**Reply by yconst** [reply]
We're working on the new version currently, and doing our best to release within the month. You can follow the development in this branch: https://github.com/tinymovr/Tinymovr/tree/sensor_support
